lib/chef/resource/rhsm_subscription.rb

```diff
@@ -32,11 +32,11 @@ class RhsmSubscription < Chef::Resource
         name_property: true
 
       action :attach, description: "Attach the node to a subscription pool." do
-        execute "Attach subscription pool #{new_resource.pool_id}" do
-          command "subscription-manager attach --pool=#{new_resource.pool_id}"
-          default_env true
-          action :run
-          not_if { subscription_attached?(new_resource.pool_id) }
+        unless subscription_attached?(new_resource.pool_id)
+          converge_by("attach subscription pool #{new_resource.pool_id}") do
+            shell_out!("subscription-manager attach --pool=#{new_resource.pool_id}")
+            flush_cache
```

Comment:
@lamont-granquist is this actually going to work? I couldn't find where `#flush_cache` is defined and have absolutely no confidence this is accurate. Well, I see that `#flush_cache` is defined in `Chef::Provider::Package::Dnf` and `Chef::Provider::Package::Yum`, but not seeing how this will be leverage in this context.

It seems like the behavior between `Chef::Provider::Package::Dnf` and `Chef::Provider::Package::Yum` is identical. Would you like me to abstract into a module and include in both `Chef::Provider::Package::Dnf` and `Chef::Provider::Package::Yum` and in this action class as well? Is there a better way here?

Reply:
Best thing to do would probably be to `build_resource(:package, "anything-goes").run_action(:flush_cache)`

Otherwise, no it doesn't work to try to call flush_cache there, that lives on an instance of the appropriate package provider and you can't get to it that way.  You could call Chef::Provider::Package::Dnf::PythonHelper.instance.restart, but that is an implementation detail of how flush_caching currently operates and I've certainly wanted it to stop doing that, so it shouldn't be considered a public API for this purpose.